design-proposals/eim-modular-decomposition.md

```diff
@@ -0,0 +1,361 @@
+deploying onboarding or adding bespoke credential bootstrapping.
+
+The current high-level architecture of EMF is illustrated below with an extended depiction of EIM components and shared infrastructure.
+
+![current high-level architecture](images/current-architecture.png)
+
+- **Inventory**: State store and reconciliation service. Exposes NB API.
+- **Host manager**: Updating Host hardware information of the fleet of Edge nodes and maintains the heartbeat
+- **Onboarding manager**: Provides mechanism to associate OS profile and onboarding workflow to be executed on the
+  Edge node
+- **Device management manager**: Provides interface to Out-of-band Device management services and platform
+  manageability agent to support Intel vPRO AMT/ISM.
+- **Maintenance manger**: Provides interface for scheduling updates (OS) on fleet of Edge nodes
+- **Networking manager**: Reconciles the state of the IP Address resource in inventory by Validating its
+  uniqueness on a per Site basis.
+- **Telemetry manager**: control plane to configure metrics/log collection from the Edge Nodes
+- **OS resource manger**: Provides mechanism to manage the OSs supported by the EIM (Does not interface with EN)
+- **LOCA manager**: Provides Lenovo Fleet manager to onboard edge nodes. Vendor manager are optional.
+- **DKAM**: Provides curation of Edge node OS and post OS install agents on Ubuntu only. Signing of ipxe and
+  uOS.
+- **CDN NGINX**: Hosts Boot artefact for EN in case of HTTPs boot and other OS artefact.
+- **Tinker bell**: Open-source component supporting OS provisioning
+- **Tenant controller**: Interfaces with Tenant manger and Inventory to enable multi-tenancy in EIM.
+- **API**: EIM Open API definition
+
+EMF also supports OXM deployment profile targeted at Equipment Manufacturers for scale provisioning of edge
+devices in factory or warehouse environments. This profile leverages the same EIM services but with a reduced
+footprint of not deploying Application and Cluster orchestration. The OXM architecture is illustrated below. It is
+apparent that the all EIM components are deployed even though the usecase only requires device OS provisioning.
+Also the deployment should EIM deployment is tightly coupled with each other and share dependencies on common
+infrastructure services such as PostgreSQL, Redis, Keycloak, and Foundation Platform Service.
+
+![current OXM architecture](images/current-oxm-architecture.png)
+
+### Modular Evolution Tracks
+
+To unlock incremental modularity without disrupting existing customers, three tracks of work are proposed:
+
+- **Track #1 (Status Quo + Use Case Enablement)**
+  - Continue leveraging the existing Argo CD Application-of-Applications pattern and our inventory plus
+    Foundation Platform Service (FPS) stack.
+  - Package “use-case specific” overlays that expose Day-0 onboarding, Day-1 configuration, and Day-2 upgrade
+    workflows via API, CLI, resource manager, and (where applicable) edge node agent bundles.
```

Comment:
Overall, the problem statement and motivations are described well but not clear on how to decompose and enable the use cases you mentioned above. It might be good to elaborate on "package use-case specific overlays".